sycl/include/sycl/multi_ptr.hpp

```diff
@@ -353,17 +352,14 @@ class __SYCL_TYPE(multi_ptr) multi_ptr {
                 std::is_same_v<RelayElementType, ElementType> &&
                 std::is_const_v<RelayElementType>>>
   operator multi_ptr<const void, Space, ConvIsDecorated>() const {
-    return multi_ptr<const void, Space, ConvIsDecorated>{detail::cast_AS<
+    return multi_ptr<const void, Space, ConvIsDecorated>{static_cast<
         typename multi_ptr<const void, Space, access::decorated::yes>::pointer>(
         get_decorated())};
   }
 
   template <access::decorated ConvIsDecorated>
   operator multi_ptr<const value_type, Space, ConvIsDecorated>() const {
-    return multi_ptr<const value_type, Space, ConvIsDecorated>{
-        detail::cast_AS<typename multi_ptr<const value_type, Space,
-                                           access::decorated::yes>::pointer>(
-            get_decorated())};
+    return multi_ptr<const value_type, Space, ConvIsDecorated>{get_decorated()};
```

Comment:
Regardless of the `access::decoration`, internal raw pointer is always decorated and that's what ctor accepts, no address cast is happening here.